Enable more precise control when using ChaCha20

joemphilips opened the pull request:
I wanted this feature for my own project. And I thought it might be worth sharing.
This adds `StreamCipherAlgorithm` class which is lower level primitives for AEAD. Currently only for ChaCha20. (See [libsodium documentation](https://libsodium.gitbook.io/doc/advanced/stream_ciphers/chacha20) for more detail )

~Mostly done. but I need to add little more tests before merging,~ ( -> Finished every tests I could think of) I just wanted a feedback before proceeding.

* Add StreamCipherAlgorithm class
* Add ChaCha20 class deriving from the above

ektrah commented:
Thanks for your contribution! (And sorry for the long pause.) I think the API needs a bit of discussion, but this code looks like a good starting point. I'd merge this now if you update to master and move the new classes in the NSec.Cryptography project to the NSec.Experimental project. Let's discuss the API in #21.

joemphilips commented:
Rebased onto master and moved classes to Experimental project.
I don't know why tests are failing in OpenSUSE. It seems that it has nothing to do with my change.
